include/jemalloc/internal/tcache_structs.h

```diff
@@ -39,8 +39,13 @@ struct tcache_slow_s {
 	szind_t		next_gc_bin;
 	szind_t		next_gc_bin_small;
 	szind_t		next_gc_bin_large;
-	/* For small bins, fill (ncached_max >> lg_fill_div). */
-	uint8_t		lg_fill_div[SC_NBINS];
+	/*
+	 * For small bins, used to calculate how many items to fill at a time.
+	 * Containing two fields, with the high 8 bits being the base, and the
+	 * low 8 bits being the offset.
+	 * The final nfill is calculated by (ncached_max >> (base - offset)).
+	 */
+	uint16_t	bin_fill_ctl[SC_NBINS];
```

Comment:
You are on the right track. Let's make this a struct (with 2 uint8_t inside) to make it as explicit as possible. That should also simplify things a bit, since you won't have to manually shift things around anymore.

Reply:
e.g. 
```
struct bin_fill_ctl_s {
  uint8 base;
  uint8 offset;
}
...
// then in tcache_slow
    bin_fill_ctl_t bin_fill_ctl_dont_access_directly[SC_NBINS];
```